### DISCUSSION TITLE
Can I make the syntax highlighter update on type instead of on save?

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [microsoft/vscode-cpptools#9407] [Q&A]
I would like to be able to get these error squiggles not just when saving - but every time I type anything into the document. Is this possible?

**Comment by michelleangela** [comment]
@ItsCubeTime
There is the setting `C_Cpp.intelliSenseUpdateDelay` which controls the delay in milliseconds before IntelliSense starts updating after a modification. You could try decreasing the delay time and see if the error squiggles updates based on your preference.

**Reply by sean-mcmanus** [reply]
What error squiggles are you referring to? IntelliSense squiggles should not require saving, but code analysis error squiggles do require saving, because it s processed from disk by the clang-tidy process.

**Reply by ItsCubeTime** [reply]
Hi, sorry for the late reply :)

> but code analysis error squiggles do require saving, because it s processed from disk by the clang-tidy process.

This is what I was referring to

**Reply by sean-mcmanus** [reply]
@ItsCubeTime Yeah, that is not currently possible, because it would require the clang-tidy process maintain all the edited files in memory and process edits, similar to how our IntelliSense process works, but it only operates only on files on disk currently. You could try filing a feature request on clang-tidy at https://github.com/llvm/llvm-project/issues, however, I'm skeptical that they would implement it, because it would be a very major change, it might not follow their design philosophy for the tool, and the slow performance of clang-tidy doesn't particularly lend itself to on the fly updating, since with large sized TUs it can take a long time to update.
